apps/meteor/client/views/admin/subscription/hooks/useCancelSubscriptionModal.spec.tsx

```diff
@@ -0,0 +1,70 @@
+import { mockAppRoot } from '@rocket.chat/mock-providers';
+import { act, renderHook, screen, waitFor } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+
+import { useCancelSubscriptionModal } from './useCancelSubscriptionModal';
+
+jest.mock('../../../../hooks/useLicense', () => ({
+	...jest.requireActual('../../../../hooks/useLicense'),
+	useLicenseName: () => ({ data: 'Starter' }),
+}));
+
+it('should open modal when open method is called', () => {
+	const { result } = renderHook(() => useCancelSubscriptionModal(), {
+		wrapper: mockAppRoot()
+			.withTranslations('en', 'core', {
+				Cancel__planName__subscription: 'Cancel {{planName}} subscription',
+			})
+			.build(),
+		legacyRoot: true,
+	});
+
+	expect(screen.queryByText('Cancel Starter subscription')).not.toBeInTheDocument();
+
+	act(() => result.current.open());
+
+	expect(screen.getByText('Cancel Starter subscription')).toBeInTheDocument();
+});
+
+it('should close modal cancel is clicked', async () => {
+	const { result } = renderHook(() => useCancelSubscriptionModal(), {
+		wrapper: mockAppRoot()
+			.withTranslations('en', 'core', {
+				Cancel__planName__subscription: 'Cancel {{planName}} subscription',
+			})
+			.build(),
+		legacyRoot: true,
+	});
+
+	act(() => result.current.open());
+	expect(screen.getByText('Cancel Starter subscription')).toBeInTheDocument();
+
+	await userEvent.click(screen.getByRole('button', { name: 'Dont_cancel' }));
+
+	expect(screen.queryByText('Cancel Starter subscription')).not.toBeInTheDocument();
+});
+
+it('should call remove license endpoint when confirm is clicked', async () => {
+	const removeLicenseEndpoint = jest.fn();
+	const { result } = renderHook(() => useCancelSubscriptionModal(), {
+		wrapper: mockAppRoot()
+			.withEndpoint('POST', '/v1/cloud.removeLicense', removeLicenseEndpoint)
+			.withLatency(100)
```

Comment:
"Test" and "latency" are two words that should never be together. Why is this necessary here?